src/MockNet/Http/Internal/HttpMessageHandler.cs

```diff
@@ -20,17 +20,19 @@ protected override async Task<SystemHttpResponseMessage> SendAsync(SystemHttpReq
         {
             var setups = this.setups.Find(request).ToList();
 
-            if (!setups.Any())
+            // there are no setups that were found successful.
+            if (!setups.Any(x => x.Exception is null))
```

Comment:
Can this be rewritten to `setups.All(x => x.Exception is MockHttpClientException)`

Reply:
Yes, that makes sense. Updated Code.